hypershift-operator/featuregate/feature.go

```diff
@@ -19,16 +23,43 @@ const (
 	OpenStack featuregate.Feature = "OpenStack"
 )
 
+var (
+	allFeatures = featuregates.NewFeatureSetAwareFeatures()
+
+	aroHCPManagedIdentitiesFeature = featuregates.NewFeature(AROHCPManagedIdentities, featuregates.WithEnableForFeatureSets(configv1.TechPreviewNoUpgrade))
```

Comment:
fwiw just a note orthogonal to the PR. While the plumbing make sense, in practice we don't even need these two in particular atm, aroHCPManagedIdentities doesn't exist anymore and openStack is basically gated implicitily by its existence in the API.

Reply:
I leave these in for now since this is pretty much just a conversion of the existing feature gates to the new format. Maybe we can create a separate ticket or something to track the need to remove stale feature gates?

Reply:
yes, let's leave it aside from that PR. I was just trying to make visible all the context since we are here